### PR TITLE
Amc 202409

### PR DESCRIPTION
**Pull request by aoifem01**
Creating GP report outputs from R, plus small formatting changes for the 2024 Autumn run.

### CODE REVIEW

File: code/x_coverage_prisons.R [hunk 1]
```diff
@@ -24,32 +24,39 @@ library(lubridate)
...
 
-rm(cutoff_date, hb_list, financial_year_due, hist_path, simd_path,
-   year1, year1_end, year1_start, year2, year2_end, year2_start)
+rm(cutoff_date, hb_list, hist_path, simd_path,
+   year1, year1_start, year2, year2_end, year2_start)
 
 financial_year <- kpi_report_years[3]
 
-cutoff_date <- "31-03-1957"
+cutoff_date <- year1_end ## AMc note: this date not verified, was 31-03-1957 when came to do 202409 autumn run
```
Comment (karen-hotopp) [inline code/x_coverage_prisons.R:40]: Come back to this. Check SPSS code and define this variable (in notes in script). Someone born 31Mar1958 is 66 today (28Oct2024).

File: code/x_master.R [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+for(x in files) {
+  
+  print(paste0("code",))
+  
+  source_file_from_subdir("code", x)
+}
+
+
+# AMC -- THIS DOESN'T WORK BECAUSE ALL THE OTHER FILES DO RM ALL
```
Comment (karen-hotopp) [inline code/x_master.R:59]: Which part doesn't work? Can we have a quick discussion re: this script?

**Review by karen-hotopp — APPROVED** [review]

File: code/write_excel/95_Source_Excel_4.R [hunk 3]
```diff
@@ -79,7 +79,7 @@ note_toc <- paste0(note_toc_data, "data for the year ending 31 March ",
 rm(note_toc_data)
 
 ## KPI 3.1 ----
-screened_year_vv <- paste0("Screened in year ending 31 March ", year_vv)
+screened_year_vvr <- paste0("Screened in year ending 31 March ", year_ww, {supsc('r')})
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/write_excel/95_Source_Excel_4.R:82]: You're definitely right, I copied and pasted and then only changed the object name oops

File: code/write_excel/93_Source_Excel_2.R [hunk 4]
```diff
@@ -165,9 +168,9 @@ rm(kpi_1.1_no_invite_before_66, kpi_1.1_invited_after_66, kpi_1.1_not_invited) #
 ### footnotes ----
 kpi_1.1_add_note1 <- paste0("1. For the previous eligible cohorts at this stage, ",
                             "the equivalent percentages of men offered screening ",
-                            "before age 66 were {x}% (", year_vv, "/", 
-                            substr(year_ww, 3,4), ") and {x}% (", year_ww, "/",
-                            substr(year_xx, 3,4), ").")
+                            "before age 66 were {x}% (", year_uu, "/",
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/write_excel/93_Source_Excel_2.R:171]: These are 2 previous financial years, so 20WW/XX is the new data (2023/24), then 20UU/VV (2021/22) and 20VV/WW (2022/23) are the 2 years prior to that, which is what this note refers to?

File: code/write_excel/96_Write_Excel_4.R [hunk 5]
```diff
@@ -47,7 +47,7 @@ template_path <- paste0("/PHI_conf/AAA/Topics/Screening/templates")
 # 2: Import data ----
 # KPI 3.1 and 3.2
 theme4_3 <- read_rds(paste0(temp_path, "/4_1_kpi_3_", yymm, ".rds"))
-table(theme4_3$kpi, theme4_3$financial_year) 
+table(theme4_3$kpi, theme4_3$fin_year)
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/write_excel/96_Write_Excel_4.R:50]: Nope everything changed to fin_year

File: code/write_excel/93_Source_Excel_2.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/write_excel/93_Source_Excel_2.R:171]: Change these back

File: code/7_4_vascular_referrals.R [hunk 6]
```diff
@@ -376,7 +385,7 @@ View(check)
 rm(check)
```
[the remark this answers is not in the document]
Comment (aoifem01) [reply code/7_4_vascular_referrals.R:385]: It gets filtered in the second instance (376) to only include cases with result_outcome in either 15 or 16, whereas there's 1 result_outcome = 20 EVAR case in the extract

**Comment by aoifem01** [comment]
@karen-hotopp just to let you know the final commit is here to be approved - once done feel free to merge and delete!

File: code/7_4_vascular_referrals.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (karen-hotopp) [reply code/7_4_vascular_referrals.R:386]: Made into issue to further investigate.

File: code/91_2_coverage_gp_practice.R [hunk 7]
```diff
@@ -135,35 +137,38 @@ make_gp_vars <- function(df, gp_lookup) {
...
   clean_names()
 
-rm(gp_history)
+rm(gp_history, gp_prac_a, gp_prac_b)
 
 
 # Flag GP practice that was relevant at the end of the financial year
+## AMc question: how do these dates relate to the 'end' of the financial year? should they be automated?
+## also, what does the { is.na(valid_from) & is.na(valid_to) > dmy("01-04-2022") } do?
+## I think this date relates to the 2021/22 FY run
```
[the remark this answers is not in the document]
Comment (karen-hotopp) [reply code/91_2_coverage_gp_practice.R:165]: Made into issue to investigate further.

**Pull request merged**